src/Inventory/Conf.php

```diff
@@ -768,6 +793,59 @@ public function showConfigForm()
         echo "</td>";
         echo "</tr>";
 
+        echo "<tr class='tab_bg_1'>";
+        echo "<th colspan=4 >" . __('Agent cleanup') . "</th></tr>";
+        echo "<tr class='tab_bg_1'>";
+        echo "<td>" . __('Update agents not have contacted server since (in days)') . "</td>";
+        echo "<td width='20%'>";
+        Dropdown::showNumber("agents_old_days", [
+            'value' => $config['agents_old_days'] ?? 0,
+            'min'   => 1,
+            'max'   => 1000,
+            'toadd' => ['0' => __('Disabled')]
+        ]);
+        echo "</td>";
+        echo "<td>" . __('Action') . "</td>";
+        echo "<td width='20%'>";
+        //action
+        $rand = Dropdown::showFromArray(
+            'agents_action',
+            [self::getActions(self::AGENT_ACTION_CLEAN), self::getActions(self::AGENT_ACTION_STATUS)],
+            ['value' => $config['agents_action'] ?? self::AGENT_ACTION_CLEAN, 'on_change' => 'changestatus();']
+        );
+        //if action == action_status => show blocation else hide blocaction
+        echo Html::scriptBlock("
+         function changestatus() {
+            if ($('#dropdown_agents_action$rand').val() != 0) {
+               $('#blocaction1').show();
+               $('#blocaction2').show();
+            } else {
+               $('#blocaction1').hide();
+               $('#blocaction2').hide();
+            }
+         }
+         changestatus();
+
+      ");
+        echo "</td>";
+        echo "</tr>";
+        //blocaction with status
+        echo "<tr class='tab_bg_1'><td colspan=2></td>";
+        echo "<td>";
+        echo "<span id='blocaction1' style='display:none'>";
+        echo __('Change the status', 'glpiinventory');
+        echo "</span>";
+        echo "</td>";
+        echo "<td width='20%'>";
+        echo "<span id='blocaction2' style='display:none'>";
+        State::dropdown(['name'   => 'agents_status',
+            'value'  => $config['agents_status'] ?? -1,
```

Comment:
Does this mean the value stored in database could be `-1` if `$config['agents_status']` is not defined or is this an UX syntax ? Imho, I expect to see `0` as default in db.

Reply:
Either one should work. It just avoids a PHP notice when the config option is not set yet. Both will default to the empty option which doesn't make sense, but it was like this in FI.

In the client request to allow both actions, the empty choice label will become "No change".